library/core/src/intrinsics/mir.rs

```diff
@@ -38,22 +37,166 @@
 //! }
 //! ```
 //!
-//! Hopefully most of this is fairly self-explanatory. Expanding on some notable details:
+//! Hopefully the syntax is fairly self-explanatory to anyone familiar with MIR. The `custom_mir`
```

Comment:
Jumping so quickly from 

> If you make a typo you’ll probably ICE.

To this vibes-based syntax documentation is kind of jarring. The docs are pretty complete now, while they don't strictly define the syntax there are a lot of examples and explanation. You can probably just drop this first sentence.